shirin_l4.py

```diff
@@ -0,0 +1,644 @@
+
+
+# Хммм, надо переделать
+
+# # GLOVE
+# 
+# Используем размерность векторного представления и ширину окна такие же, что и в word2vec. Счетчик совместных вхождений у нас будет с долями - чем дальше находятся слова, тем меньше увеличивается счетчик по формуле $^1/_{ dist}$, структура графа подсмотрена на [гитхабе](https://github.com/GradySimon/tensorflow-glove/blob/master/tf_glove.py)
+
+# In[14]:
+
+cont_width = 4
+embedding_size = 200
+
+
+# In[14]:
+
+batch_size = 100
+
+
+# In[15]:
+
+X = sparse.lil_matrix((VOC_SIZE, VOC_SIZE), dtype=np.float64)
+for i in range(len(data)):
+    for j in range(max(0, i - cont_width), i):
+        X[data[i], data[j]] += 1.0 / (i - j)
+    for j in range(min((i + 1), len(data)), min((i + 1 + cont_width), len(data))):
+        X[data[i], data[j]] += 1.0 / (j - i)
+X = sparse.coo_matrix(X)
+
+
+# In[16]:
+
+GAMMA = 0.05
+
+
+# In[19]:
+
+glove_graph = tf.Graph()
+with glove_graph.as_default():
+
+    focal_input = tf.placeholder(tf.int32, shape=[batch_size],
+                                        name="focal_words")
+    context_input = tf.placeholder(tf.int32, shape=[batch_size],
+                                          name="context_words")
+    cooccurrence_count = tf.placeholder(tf.float32, shape=[batch_size],
+                                               name="cooccurrence_count")
+
+    focal_embeddings = tf.Variable(
+        tf.random_uniform([VOC_SIZE, embedding_size], 1.0, -1.0) / float(embedding_size + 1))
+    context_embeddings = tf.Variable(
+        tf.random_uniform([VOC_SIZE, embedding_size], 1.0, -1.0) / float(embedding_size + 1))
+
+    focal_biases = tf.Variable(tf.random_uniform([VOC_SIZE], 1.0, -1.0) / float(embedding_size + 1))
+    context_biases = tf.Variable(tf.random_uniform([VOC_SIZE], 1.0, -1.0) / float(embedding_size + 1))
+
+    focal_embedding = tf.nn.embedding_lookup([focal_embeddings], focal_input)
+    context_embedding = tf.nn.embedding_lookup([context_embeddings], context_input)
+    focal_bias = tf.nn.embedding_lookup([focal_biases], focal_input)
+    context_bias = tf.nn.embedding_lookup([context_biases], context_input)
+
+    f = tf.minimum(1.0, tf.pow(tf.div(cooccurrence_count, 400),0.75))
+
+    embedding_product = tf.reduce_sum(tf.multiply(focal_embedding, context_embedding), 1)
+
+    log_cooccurrences = tf.log(tf.to_float(cooccurrence_count))
+
+    distance_expr = tf.square(tf.add_n([
+        embedding_product,
+        focal_bias,
+        context_bias,
+        tf.negative(log_cooccurrences)]))
+
+    single_losses = tf.multiply(f, distance_expr)
+    total_loss = tf.reduce_sum(single_losses)
+    optimizer = tf.train.AdagradOptimizer(GAMMA).minimize(total_loss)
+
+    combined_embeddings = tf.add(focal_embeddings, context_embeddings,
+                                        name="combined_embeddings")
+
+
+# In[20]:
+
+num_epochs = 12
+
+
+# In[21]:
+
+with tf.Session(graph=glove_graph) as session:
+    tf.initialize_all_variables().run()
```

Comment:
`WARNING:tensorflow:From <ipython-input-69-38d746dc1320>:2 in <module>.: initialize_all_variables (from tensorflow.python.ops.variables) is deprecated and will be removed after 2017-03-02.
Instructions for updating:
Use `tf.global_variables_initializer` instead.`